### PR TITLE
InsightLOCOTest Refactor 

### PR DESCRIPTION
**Pull request by AdamChit**
**Related issues**
Long test that needs to be refactored with the main goal of making it more readable and breaking it up into separate tests

**Describe the proposed solution**
First Change from FlatSpec to [FunSpec](http://doc.scalatest.org/1.7.1/org/scalatest/FunSpec.html) then break up the test into 5 different test cases 

**Describe alternatives you've considered**
could of put the data generation code into a function, but generating data is expensive so we wouldn't want to make repeated calls to it

### CODE REVIEW

File: core/src/test/scala/com/salesforce/op/stages/impl/insights/RecordInsightsLOCOTest.scala [hunk 1]
```diff
@@ -404,10 +411,8 @@ class RecordInsightsLOCOTest extends FlatSpec with TestSparkContext with RecordI
    *
    * @param predicate  predicate used by RecordInsights in order to aggregate
    */
-  private def assertAggregatedWithPredicate(
-    predicate: OpVectorColumnHistory => Boolean,
-    testData: RecordInsightsTestData[LogisticRegressionModel]
-  ): Unit = {
+  private def assertAggregatedWithPredicate(predicate: OpVectorColumnHistory => Boolean,
+                                            testData: RecordInsightsTestData[LogisticRegressionModel]): Unit = {
```
Comment (crupley) [inline core/src/test/scala/com/salesforce/op/stages/impl/insights/RecordInsightsLOCOTest.scala]: Can you revert the formatting here to the way it was before to match scala style guidelines?

File: core/src/test/scala/com/salesforce/op/stages/impl/insights/RecordInsightsLOCOTest.scala [hunk 2]
```diff
@@ -202,7 +202,8 @@ class RecordInsightsLOCOTest extends FlatSpec with TestSparkContext with RecordI
     }
   }
 
-  it should "return the most predictive features for data generated with a strong relation to the label" in {
+  describe("data strongly related to label. ") {
```
Comment (crupley) [inline core/src/test/scala/com/salesforce/op/stages/impl/insights/RecordInsightsLOCOTest.scala]: This description doesn't seem to match the following section. Can you add something more descriptive?

File: core/src/test/scala/com/salesforce/op/stages/impl/insights/RecordInsightsLOCOTest.scala [hunk 3]
```diff
@@ -96,7 +96,7 @@ class RecordInsightsLOCOTest extends FlatSpec with TestSparkContext with RecordI
     )
   )
   // scalastyle:on
-  Spec[RecordInsightsLOCO[_]] should "work with randomly generated features and binary logistic regression" in {
+  it ("should work with randomly generated features and binary logistic regression") {
```
Comment (tovbinm) [inline core/src/test/scala/com/salesforce/op/stages/impl/insights/RecordInsightsLOCOTest.scala]: replace `it` with `Spec[RecordInsightsLOCO[_]]`

Comment (AdamChit) [reply core/src/test/scala/com/salesforce/op/stages/impl/insights/RecordInsightsLOCOTest.scala]: Can't directly do that with FunSpec, what worked best was wrapping `describe(Spec[RecordInsightsLOCO[_]])`  around the tests.

What the updated tests look like:

Screen Shot 2019-10-07 at 10 32 14 AM

**Comment by AdamChit** [comment]
code 
https://codecov.io/gh/salesforce/TransmogrifAI/compare/a83bf4639b2a9324ae27d1d49edf66a6245f6ac0...a4a0e653b6d1437dbef7a3b388baa740834c0b58/changes

**Pull request merged**